### PR TITLE
Ivy perf updates 08 02

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

File: packages/core/src/render3/instructions/shared.ts [hunk 1]
```diff
@@ -407,22 +395,20 @@ export function renderEmbeddedTemplate<T>(viewToRender: LView, tView: TView, con
   const _isParent = getIsParent();
   const _previousOrParentTNode = getPreviousOrParentTNode();
   let oldView: LView;
-  if (viewToRender[FLAGS] & LViewFlags.IsRoot) {
-    // This is a root view inside the view tree
-    tickRootContext(getRootContext(viewToRender));
```
Comment (AndrewKushnir) [inline packages/core/src/render3/instructions/shared.ts]: @pkozlowski-opensource just curious why this call is no longer needed? Thank you.

Comment (pkozlowski-opensource) [reply packages/core/src/render3/instructions/shared.ts]: So what is going on here is that a `LContainer` can have 2 types of views:
* embedded views (created from `<ng-template>`;
* root component views (created from dynamically instantiated components).

it turns out that `refreshDescendantViews` knows how to deal with "root component views " so we don't need any special code paths here. Generally speaking there is _lots_ of code / logic duplication when it comes to refreshing views - it seems like it grown "organically" - but in many cases we could do with just 2 generic functions:
* process view in a creation mode;
* process view in the update mode.
I'm working on more refactorings that will clean up even more special-case code path, just taking baby steps here.

Comment (AndrewKushnir) [reply packages/core/src/render3/instructions/shared.ts]: Thanks for additional info @pkozlowski-opensource!

File: packages/core/src/render3/instructions/shared.ts [hunk 2]
```diff
@@ -1480,7 +1479,7 @@ export function createLContainer(
  * them
  * by executing an associated template function.
  */
-function refreshDynamicEmbeddedViews(lView: LView) {
+export function refreshDynamicEmbeddedViews(lView: LView) {
```
Comment (mhevery) [inline packages/core/src/render3/instructions/shared.ts]: The `export` does not seem to be needed here.

**Review by mhevery — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
- [VE Presubmit](http://test/OCL:261390449:BASE:261390478:1564780805738:a10a4d1d)
- [Ivy Presubmit](http://test/OCL:261390449:BASE:261390672:1564780870471:4ef9daf7)

**Comment by AndrewKushnir** [comment]
FYI, VE and Ivy Presubmits look normal 👍

**Pull request closed without merging**